crates/rust-analyzer/src/config.rs

```diff
@@ -1565,6 +1613,14 @@ impl Config {
                             .ok()
                             .map(Into::into)
                     }
+                    ManifestOrProjectJson::DiscoveredProjectJson { data, buildfile } => {
+                        let root: &AbsPath =
+                            buildfile.as_path().try_into().expect("unable to get AbsPathBuf");
```

Comment:
Feels like `DiscoveredProjectJson` should contain an `AbsPathBuf` instead, having the validation done somewhere up the call chain

Reply:
yes, but unfortunately, the `DiscoveredProjectJson` contains a `project_model::ProjectJsonData`, which currently needs (I think?) to have non-absolute paths so that we can emit SCIP Protobufs to [Glean](https://glean.software/docs/indexer/lsif-rust/) for code navigation in web browsers.

Reply:
oh, i misunderstood the question. yeah, this can be an `AbsPathBuf`.

Reply:
just pushed a change that moved over to `AbsPathBuf` for the buildfile, but this required adding `#[serde(serialize_with = "..."]`/`#[serde(deserialize_with = "..."]` annotations to avoid making `AbsPath`/`AbsPathBuf` implement `Serialize`/`Deserialize`. I'd like to avoid implementing `Serialize`/`Deserialize` on `AbsPathBuf` because of this comment: https://github.com/rust-lang/rust-analyzer/blob/1b283db47f8de1412c851c92bb4ce4ef039ff8ff/crates/paths/Cargo.toml#L16-L18

I think hack could go away if/when `Config::linked_projects` isn't used as a source of truth anymore.